app/feedme/parsers/pdf_parser.py

```
@@ -0,0 +1,670 @@
+                return PDFParseResult(
+                    success=True,
+                    pages=pages,
+                    metadata=metadata,
+                    total_text=total_text,
+                    total_chars=total_chars,
+                    processing_time_ms=0,  # Will be set by async wrapper
+                    warnings=warnings if warnings else None
+                )
+                
+        except Exception as e:
+            raise Exception(f"Failed to extract text: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)